mksh: Added. Fixes #448

tox-wtf opened the pull request.

tox-wtf committed:
mksh: Added. Fixes #448

zeckma-tech commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,128 @@
+    </para>
+
+<screen><userinput>sh Build.sh -r
+</userinput></screen>
```
Put the ending tags at the end of the above line (51).

zeckma-tech commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,128 @@
+
+<screen role="root"><userinput>install -vDm755 mksh -t /usr/bin/
+install -vDm644 mksh.1 -t /usr/share/man/man1/
+</userinput></screen>
```
Put the ending tags at the end of the above line (63).

tox-wtf committed:
Move ending tags

zeckma-tech commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,126 @@
+      Now as the &root; user:
+    </para>
+
+<screen role="root"><userinput>install -vDm755 mksh -t /usr/bin/
```
Add ` &amp;&amp;` at the end of `mksh -t /usr/bin/`.

renodr commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,126 @@
+
+    <para>
+      The mksh package provides the MirBSD Korn Shell, a small, fast, reliable
+      shell suitable for both scripting and interactive use.
```
I'd like to request a minor reword here, can we say "The mksh package provides the MirBSD Korn Shell, which is a small, fast, and reliable shell suitable for both scripting and interactive use." instead? I like how your description but I think this would flow a little bit better

zeckma-tech replied:
Agreed.

tox-wtf committed:
Resolve requested changes

zeckma-tech commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,126 @@
+<screen><userinput>sh Build.sh -r</userinput></screen>
+
+    <para>
+      To test the results, issue: <command>./test.sh</command>
```
Add `.` at the end of `<command>./test.sh</command>`, like so: `<command>./test.sh</command>.`

tox-wtf replied:
I also removed the colon.

zeckma-tech commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,126 @@
+  <!ENTITY % general-entities SYSTEM "../../general.ent">
+  %general-entities;
+
+  <!ENTITY mksh-download-http "https://github.com/MirBSD/mksh/archive/mksh-&mksh-version;.tar.gz">
```
Please adjust this link to ensure that `mksh-R59c.tar.gz` is the filename and `mksh-R59c` is the extracted directory. As is, the extracted directory ends up being `mksh-mksh-R59c`. This causes issues for users using auto-complete tabbing.

tox-wtf replied:
I don't believe this is something I can fix since upstream includes the name in the tag.

tox-wtf replied:
If there's an adjusted version of the link that does resolve this, please share it.

zeckma-tech replied:
I'd personally use `https://mbsd.evolvis.org/MirOS/dist/mir/mksh/mksh-&mksh-version;.tgz`. The extracted directory ends up being `mksh/` but I think it's better and pulls from MirBSD directly instead of a Github mirror.

tox-wtf committed:
Formatting fix

zeckma-tech commented on general/shells/mksh.xml:
```diff
@@ -0,0 +1,126 @@
+<screen><userinput>sh Build.sh -r</userinput></screen>
+
+    <para>
+      To test the results, issue <command>./test.sh</command>.
```
Just to be safe, for the command, I'd issue `./mksh test.sh` to ensure that the recently built `mksh` is used. I did read the test script and seems to use mksh in the end, but I think it's just best practice to ensure that `mksh` is used here.

tox-wtf committed:
Resolve requested changes

- Don't use the GitHub mirror
- Call `test.sh` with `./mksh`
- Also align whitespace for `install`s
